### ISSUE TITLE
No deffile extracted with inspect -d

### ISSUE THREAD

**Issue by phupe** [apptainer/apptainer#521]
### Version of Apptainer

Apptainer version installed from main branch

```
apptainer --version
apptainer version 1.0.2+349-gdf2e94880
```

### Expected behavior

`apptainer inspect -d` must output the deffile.

### Actual behavior

Output is empty when using sif image built with apptainer while it works if the sif file has been built with former singularity version. 

### Steps to reproduce this behavior

```sh
#! /bin/bash

set -oue

### build image with singularity 3.8.4
 sudo /opt/singularity/singularity-3.8.4/bin/singularity build alpine-singularity3.8.4.sif docker://alpine

### build image with apptainer
 sudo /opt/apptainer/apptainer/bin/apptainer build alpine-apptainer.sif docker://alpine

### extract deffile with inspect
/opt/singularity/singularity-3.8.4/bin/singularity inspect -d alpine-singularity3.8.4.sif   # OK
/opt/singularity/singularity-3.8.4/bin/singularity inspect -d alpine-apptainer.sif          # KO -> no error but no deffile extracted
/opt/apptainer/apptainer/bin/apptainer inspect -d alpine-singularity3.8.4.sif               # OK
/opt/apptainer/apptainer/bin/apptainer inspect -d alpine-apptainer.sif                      # KO -> no error but no deffile extracted

```

### What OS/distro are you running

```sh
$ cat /etc/os-release
Ubuntu 22.04 LTS
```

### How did you install Apptainer

From source using main branch.

**Comment by DrDaveD** [comment]
Thanks for the report.  I confirm the same behavior with singularity-3.8.7 (built images work) and apptainer-1.0.2 (built images don't work).  Because that functionality is based on the sif library (I assume, didn't check that), I also checked singularity-ce-3.9.2 and -3.9.10, but they do not have the problem.

**Comment by DrDaveD** [comment]
Fixed in #527.
